### DISCUSSION TITLE
Limit on the zoom depth and limits on the x and y axes

### DISCUSSION THREAD

**Discussion by AlexJagello** [GSI-CS-CO/chart-fx#468] [Q&A] [answered]
Hello. Great thanks for the nice product.
i'm beginner in chartfx. I'd like to add a limit of the zoom depth and limits of the x and y axes. Is it possible to do this using standard plugin? And how?
Thanks in advance!

**Comment by wirew0rm** [comment, marked as answer]
This is not supported at the moment, but depending on your requirements shouldn't be too hard to implement.

There are multiple ways to influence the axes ranges:
- interactive using the Zoomer plugin,
- Programmatically using axis.set(min, max),
- Using auto-ranging axes to determine the range from the datasets shown in the chart.

For the first one, you would have to extend the Zoomer plugin and check the zoom range against your limits. The second one is probably not relevant. For limiting auto-ranging you would have to extend the DefaultNumericAxis to add properties to clamp the auto ranging. You could use these axis properties of your "LimitedNumericAxis" in your "LimitiedZoomer". But as a start I would recommend to start with the simplest extension of the Zoomer plugin and add other functionality later on.
